app/controllers/user_controller.rb

```diff
@@ -0,0 +1,23 @@
+class UserController < ApplicationController
+  def update
+    flash.clear
+    @user = User.find_by(id: session[:current_user_id])
+    if @user.nil?
+      redirect_to login_path, flash: { :error => "Please log-in!" }
+      return
+    end
+    first_name = params[:user][:first_name]
+    last_name = params[:user][:last_name]
+    email = params[:user][:email]
+    sid = params[:user][:sid]
+    if sid.blank? or sid.length != 10
```

Comment:
I updated the checks for first_name, last_name, etc. to be validated in the User model. Now I only need to check that the sid entered isn't blank (nil or empty) and that the length is valid.